### ISSUE TITLE
Exit code 0 after emitting useless suppression warning

### ISSUE THREAD

**Issue by l0b0** [PyCQA/pylint#8456]
### Bug description

The following code has a useless suppression:

```python
object() # pylint: disable=abstract-class-instantiated
```

After enabling the useless suppression rule pylint prints the warning but exits with success.

### Configuration

```ini
[tool.pylint.MASTER]
disable = [
    "missing-module-docstring"
]
enable = [
    "useless-suppression"
]
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
$ pylint test.py
************* Module test
test.py:1:0: I0021: Useless suppression of 'abstract-class-instantiated' (useless-suppression)

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

$ echo $?
0
```

### Expected behavior

The exit code should be non-zero, as per [the documentation](https://pylint.readthedocs.io/en/v2.17.0/user_guide/usage/run.html#exit-codes).

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.10.9 (main, Dec  6 2022, 18:44:57) [GCC 11.3.0]
```

### OS / Environment

NixOS 22.11

### Additional dependencies

astroid==2.15.0
dill==0.3.6
isort==5.12.0
lazy-object-proxy==1.9.0
mccabe==0.7.0
platformdirs==3.1.1
pylint==2.17.0
tomli==2.0.1
tomlkit==0.11.6
typing_extensions==4.5.0
wrapt==1.15.0

**Comment by l0b0** [comment]
I'd missed [`fail-on="I"`](https://github.com/Polyconseil/pylint-strict-informational#pylint-strict-informational).
